<a href="https://colab.research.google.com/github/flaviohahn/ApMaquina/blob/master/Projeto_AM_Dem%C3%A9trios_Fl%C3%A1vio_Mauric%C3%A9sar.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Criando o Dataset**

In [ ]:
#Realizando as importações
import pandas as pd
import numpy as np

In [ ]:
#Criando o dataset:
url = 'https://raw.githubusercontent.com/flaviohahn/ApMaquina/master/bank-full.csv'
dados = pd.read_csv(url, sep=';')
dados.head(5)

age           job  marital  education  ... pdays  previous poutcome   y
0   58    management  married   tertiary  ...    -1         0  unknown  no
1   44    technician   single  secondary  ...    -1         0  unknown  no
2   33  entrepreneur  married  secondary  ...    -1         0  unknown  no
3   47   blue-collar  married    unknown  ...    -1         0  unknown  no
4   33       unknown   single    unknown  ...    -1         0  unknown  no

[5 rows x 17 columns]

In [ ]:
dados.shape

(45211, 17)

O dataset possui 17 colunas e 45211 linhas

Colunas presentes no dataset

In [ ]:
dados.columns

Index(['age', 'job', 'marital', 'education', 'default', 'balance', 'housing',
       'loan', 'contact', 'day', 'month', 'duration', 'campaign', 'pdays',
       'previous', 'poutcome', 'y'],
      dtype='object')

Contagem dos numeros nulos

In [ ]:
dados.isnull().sum()

age          0
job          0
marital      0
education    0
default      0
balance      0
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
y            0
dtype: int64

Contagem de dados não nulos

In [ ]:
dados.count()

age          45211
job          45211
marital      45211
education    45211
default      45211
balance      45211
housing      45211
loan         45211
contact      45211
day          45211
month        45211
duration     45211
campaign     45211
pdays        45211
previous     45211
poutcome     45211
y            45211
dtype: int64

Não foram encontrados dados nulos, no entanto existe a presença de dados desconhecidos 'unknown'


**Limpeza**

Entre as classes que possuem dados desconhecido temos:

In [ ]:
sum(dados.job == 'unknown')


288

In [ ]:
sum(dados.poutcome == 'unknown')

36959

In [ ]:
sum(dados.education == 'unknown')

1857

In [ ]:
sum(dados.contact == 'unknown')

13020

**As demais classes não apresentaram valores desconhecidos 'unknown'**

**Desta forma optamos por excluir as colunas 'contact', 'poutcome' pois tais dados desconhecidos podem influenciar diretamente nos resultados **

In [ ]:
df = dados.drop(columns=['contact', 'poutcome'])
df.head()

age           job  marital  education  ... campaign  pdays previous   y
0   58    management  married   tertiary  ...        1     -1        0  no
1   44    technician   single  secondary  ...        1     -1        0  no
2   33  entrepreneur  married  secondary  ...        1     -1        0  no
3   47   blue-collar  married    unknown  ...        1     -1        0  no
4   33       unknown   single    unknown  ...        1     -1        0  no

[5 rows x 15 columns]

**Assim temos a seguinte descrição:**

In [ ]:
df.describe()

age        balance  ...         pdays      previous
count  45211.000000   45211.000000  ...  45211.000000  45211.000000
mean      40.936210    1362.272058  ...     40.197828      0.580323
std       10.618762    3044.765829  ...    100.128746      2.303441
min       18.000000   -8019.000000  ...     -1.000000      0.000000
25%       33.000000      72.000000  ...     -1.000000      0.000000
50%       39.000000     448.000000  ...     -1.000000      0.000000
75%       48.000000    1428.000000  ...     -1.000000      0.000000
max       95.000000  102127.000000  ...    871.000000    275.000000

[8 rows x 7 columns]

**Também aplicamos o comando a seguir para excluir as linhas com dados desconhecidos 'unknown' nas colunas job e educatios**

In [ ]:
df = df.drop(df[df.job == 'unknown'].index)
df = df.drop(df[df.education == 'unknown'].index)


In [ ]:
sum(df.job == 'unknown')
sum(df.education == 'unknown')


0

**Tendo como base para o trabalho o seguinte dataset:**

In [ ]:
df.head()

age           job  marital  education  ... campaign  pdays previous   y
0   58    management  married   tertiary  ...        1     -1        0  no
1   44    technician   single  secondary  ...        1     -1        0  no
2   33  entrepreneur  married  secondary  ...        1     -1        0  no
5   35    management  married   tertiary  ...        1     -1        0  no
6   28    management   single   tertiary  ...        1     -1        0  no

[5 rows x 15 columns]

**Passamos a ter 43193 linhas e 15 colunas**

In [ ]:
df.shape

(43193, 15)

In [ ]:
df.describe()

age        balance  ...         pdays      previous
count  43193.000000   43193.000000  ...  43193.000000  43193.000000
mean      40.764082    1354.027342  ...     40.404070      0.584863
std       10.512640    3042.103625  ...    100.420624      2.332672
min       18.000000   -8019.000000  ...     -1.000000      0.000000
25%       33.000000      71.000000  ...     -1.000000      0.000000
50%       39.000000     442.000000  ...     -1.000000      0.000000
75%       48.000000    1412.000000  ...     -1.000000      0.000000
max       95.000000  102127.000000  ...    871.000000    275.000000

[8 rows x 7 columns]

In [ ]:
dados.columns

Index(['age', 'job', 'marital', 'education', 'default', 'balance', 'housing',
       'loan', 'contact', 'day', 'month', 'duration', 'campaign', 'pdays',
       'previous', 'poutcome', 'y'],
      dtype='object')